### ISSUE TITLE
Add abstract getText() method to the TextObject class 

### ISSUE THREAD

**Issue by janjanneck** [slackapi/java-slack-sdk#759]
Hey!

I'm currently running into an issue where I am testing the text of `OptionObject`s returned by a `StaticSelectElement`. 

To get the actual String representing the text of an `OptionObject`, I have to know (or test for) their specific implementation and then cast the class to either `MarkdownTextObject`, `PlainTextObject`, or `UnknownTextObject` since their parent class has no method that allows me to retrieve their text value directly. 

I suggest modifying `TextObject` to include the declaration of `public abstract String getText();`

Additionally, the naming of `OptionObject`'s `getText()` method has tripped me up several times since `getText` suggests a String as the return value. `getTextObject()` appears to be a more fitting name for this method.  

This would allow the following code to work without ugly casts in between the maps:

```java
List<String> textOptions = options.stream()
        .map(OptionObject::getTextObject)
        .map(TextObject::getText)
        .collect(Collectors.toList());
```

Testing locally worked without issues. I could submit the PR if you consider this to be feasible.

### Category

* [ ] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [x] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

**Comment by seratch** [comment]
Hi @janjanneck, thanks for taking the time to write in!

This is a great suggestion. Perhaps, we can add not only `String getText()` but also `String getType()`. The `UnknownTextObject` can return null for both (or either of them).

We'll add the fix in the next minor version (the release date is not yet decided but before long)

**Comment by janjanneck** [comment]
The `getType()` method could even be declared on the `ContextBlockElement` interface.

**Comment by seratch** [comment]
@janjanneck Thanks, that's true! I appreciate it 👍

**Comment by janjanneck** [comment]
And while we are at it: All classes implementing `LayoutBlock` also declare the `getBlockId()` method which should become part of the interface.

**Comment by seratch** [comment]
👋 v1.10.0-RC1 (release candidate) is now available in the Maven Central repositories! Try the version out if you have a chance to see the issue is resolved 🙇 We'll release v1.10.0 within a few days.
